tests/micropython/ringio_big.py

```diff
@@ -0,0 +1,32 @@
+# Check that micropython.RingIO works correctly.
+
+import micropython
+
+try:
+    micropython.RingIO
+except AttributeError:
+    print("SKIP")
+    raise SystemExit
+
+try:
+    # Ensure the heap is big enough for the test.
+    bytearray(70000)
```

Comment:
I'm not sure this is going to work.

It could pass, yet the below allocations fail, due to the GC being conservative.  Or due to fragmentation that happens after this allocation succeeds.

I suggest just wrapping the whole thing below in try/except MemoryError and skipping if any of the parts fail.  That's not ideal, but the main thing here is to pass to get coverage, which will always be the case on the unix port.

Reply:
updated again. is this what you're lookning for?